test/jdk/jdk/jfr/tool/TestMetadata.java

```diff
@@ -75,4 +87,65 @@ public static void main(String[] args) throws Throwable {
             lineNumber++;
         }
     }
-}
+
+    static void testEventTypeNum() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
+            }
+        }
+        List<EventType> eventTypes = FlightRecorder.getFlightRecorder().getEventTypes();
+        List<String> expectedNames = new ArrayList<>();
+        for (EventType eventType : eventTypes) {
+            expectedNames.add(eventType.getName());
+        }
+        Asserts.assertGTE(eventNames.size(), expectedNames.size());
+    }
+
+    static void testDeterministic() throws Throwable {
+        OutputAnalyzer output = ExecuteHelper.jfr("metadata", "--events", "CPULoad,GarbageCollection");
+        List<String> eventNames = new ArrayList<>();
+        List<String> lines = output.asLines();
+
+        for (String line : lines) {
+            if (line.startsWith("@Name(\"")) {
+                eventNames.add(line.substring(7, line.indexOf("\"", 7)));
+            }
+        }
+        Asserts.assertGTE(eventNames.size(), 2);
+    }
+
+    static void testWildcardAndAcronym() throws Throwable {
```

Comment:
The name of the seems misleading since it doesn't test acronyms, testWildcard() shuld be sufficient.

Reply:
`MyEv*` and `Custo*` are expected to acronym filter, so I keep this name, please let me know if I did wrong.